### ISSUE TITLE
In step 5, strange IDE errors / warnings pop up

### ISSUE THREAD

**Issue by pluppplupp** [Starcounter/InvoiceDemo#43]

**Comment by miyconst** [comment]
Could you please provide some more info?

**Comment by pluppplupp** [comment]
I can try to do the tutorial once more, to check if I get the same behaviour again..

**Comment by pluppplupp** [comment]
I've tried it again, and figured out how it works. I have ReSharper installed. ReSharper is showing these errors. When disabling ReSharper it works as expected. When ReSharper is enabled the errors above occurs until you save the file with the class extending Page.

**Comment by pluppplupp** [comment]
Resharper version Ultimate 2015.2 (9.2)

**Comment by miyconst** [comment]
Okay, this is something @chrhol and @per-samuelsson should take a look at. The problem is that we have to generate (weave) some c# code for `.json` files, sometimes this cause errors, when generated code is not updated according to `json` file on time.

Personally I dislike Resharper, but I know that many people like it, and we should support it.

**Comment by pluppplupp** [comment]
Yes, since the tooling in VS.NET is getting better and better, ReSharper will be less used I think. 

ReSharper also tells me to remove the Page class since it exists on the generated partial.

![2015-12-14_11-50-23](https://cloud.githubusercontent.com/assets/1765023/11779114/70abda44-a259-11e5-911d-ea8a8508101a.png)

**Comment by per-samuelsson** [comment]
>  The problem is that we have to generate (weave) some c# code for .json files,

Just a super-minor comment on this: 

*Weaving* happen at the time you start an application, and works on the already compiled code. Its a black-box and nothing that can cause these kind of things in the IDE. 

The *generation of C# code* based on JSON is something completely else, and is driven by `msbuild`, compiling the generated classes on the fly, everytime you modify the JSON or the code behind.

For the end user, like @pluppplupp, it's of course absolutly not interesting at all, but anyway. :smile:

**Comment by per-samuelsson** [comment]
> ReSharper also tells me to remove the Page class since it exists on the generated partial.

This seem to be out of our control. I mean, C# support this (or even promote it) when defining partial classe. Why ReSharper complain about it, I don't understand.

There are really old issues about this - [one](https://devnet.jetbrains.com/message/5181004) and [two](https://devnet.jetbrains.com/message/5218793) - and one if them even seem to be acknoledged. But maybe they never addressed it.
